orchagent/port/porthlpr.cpp

```diff
@@ -697,6 +696,27 @@ bool PortHelper::parsePortLinkTraining(PortConfig &port, const std::string &fiel
     return true;
 }
 
+// custom_collection specialization
+bool PortHelper::parsePortSerdes(
+    decltype(PortSerdes_t::custom_collection)& serdes,
+    const std::string& field,
+    const std::string& value) const
+{
+    SWSS_LOG_ENTER();
+
+    if (value.empty())
+    {
+        SWSS_LOG_ERROR("Failed to parse field(%s): empty string is prohibited",
+                       field.c_str());
+        return false;
+    }
+
+    // directly assign the raw string
+    serdes.value  = value;
```

Comment:
@longhuan-cisco I don't see a reason to NOT use existing parsePortSerdes() where serdes.value is a vector of uint32_t. This new function keeps the serdes.value as a string of values.

Reply:
> @longhuan-cisco I don't see a reason to NOT use existing parsePortSerdes() where serdes.value is a vector of uint32_t. This new function keeps the serdes.value as a string of values.

Keep only the existing parsePortSerdes() and use `std::enable_if` to achieve different implementations for different types of `serdes.value`

> @longhuan-cisco How would the std::string& value look like? Is this a json string? Any example?

It's a JSON,  e,g. `'{"attributes":[{"XYZ":{"value":[12,13]}},{"ABC":{"value":[3,4]}}]}'`